src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -205,10 +235,75 @@ export class ExamExerciseImportComponent implements OnInit {
+     * @param exerciseGroup exercise group of the exercise
+     * @param checkForTitle true if the title should be checked, otherwise the short name is checked
+     */
+    checkForDuplicatedTitlesOrShortNames(exercise: Exercise, exerciseGroup: ExerciseGroup, checkForTitle: boolean) {
+        if (!this.exerciseIsSelected(exercise, exerciseGroup)) {
+            return;
+        }
+
+        const duplicatesToCheck = checkForTitle ? this.exercisesWithDuplicatedTitles : this.exercisesWithDuplicatedShortNames;
+        let hasDuplicate = false;
+        this.selectedExercises.forEach((exerciseGroup) => {
+            exerciseGroup.forEach((ex) => {
+                if (ex.type === ExerciseType.PROGRAMMING && ex !== exercise) {
+                    if ((checkForTitle && ex.title === exercise.title) || (!checkForTitle && ex.shortName === exercise.shortName)) {
+                        hasDuplicate = true;
+                        duplicatesToCheck.set(ex.id!, checkForTitle ? exercise.title! : exercise.shortName!);
+                    }
+                }
+            });
+        });
+
+        //check if the exercise was a duplicate before
+        const titleOrShortName = duplicatesToCheck.get(exercise.id!);
+        if (duplicatesToCheck.delete(exercise.id!)) {
+            this.removeExerciseFromDuplicates(titleOrShortName!, checkForTitle);
+        }
+
+        if (hasDuplicate) {
+            duplicatesToCheck.set(exercise.id!, checkForTitle ? exercise.title! : exercise.shortName!);
+        }
+    }
+
+    /**
+     * after a duplicate programming exercise's title / short name has been edited or a programming exercise has been unselected,
+     * other duplicates need to be checked if they can be removed from the map. I case there is only one exercise with the same title / short name
+     * it is no longer considered a duplicate and is removed
+     * @param titleOrShortName title / short name of the exercise that was a duplicate before
+     * @param checkForTitle   true if the exercise title duplication should be checked, otherwise short name duplication is checked
+     *
+     */
+    removeExerciseFromDuplicates(titleOrShortName: string, checkForTitle: boolean) {
+        const setToCheck = checkForTitle ? this.exercisesWithDuplicatedTitles : this.exercisesWithDuplicatedShortNames;
+        let stillADuplicate = false;
+        const map = new Map<string, number>();
+
+        setToCheck.forEach((value, key) => {
+            if (value === titleOrShortName) {
+                if (!map.has(titleOrShortName)) {
+                    map.set(titleOrShortName, key);
+                } else {
+                    // more than one duplicate has been found
+                    stillADuplicate = true;
+                    return;
+                }
+            }
+        });
+        if (!stillADuplicate) {
+            map.forEach((element) => setToCheck.delete(element));
+        }
+    }
```

Comment:
I'm a bit unhappy with this function because it seems possible to simplify it by a lot. Could you re-generate the corresponding sets instead? Or simplify this in another way? I didn't check it enough to be 100% certain, but I believe this code to be equivalent to this, for example:
```
    const setToCheck = checkForTitle ? this.exercisesWithDuplicatedTitles : this.exercisesWithDuplicatedShortNames;

    const filteredKeys = Array.from(setToCheck.keys()).filter(key => setToCheck.get(key) === titleOrShortName);

    if (filteredKeys.length <= 1) {
        filteredKeys.forEach(key => setToCheck.delete(key));
    }
```

Reply:
I really like your solution! I tested it out and it seems to work well. Thanks :)